src/praisonai-agents/praisonaiagents/llm/llm.py

```
@@ -5643,6 +5649,47 @@ def _detail_value(detail_names: tuple[str, ...], name: str) -> int:
+                if choices:
+                    choice = choices[0]
+                    finish_reason = getattr(choice, "finish_reason", None)
+                    msg = getattr(choice, "message", None)
+                    refusal = getattr(msg, "refusal", None) if msg is not None else None
+            if finish_reason is None and not refusal:
+                return
+            from ..agent.run_outcome import classify_finish_reason
+            reason = classify_finish_reason(finish_reason, refusal)
+            if reason is not None and self._last_stop_reason == "completed":
+                self._last_stop_reason = reason
+        except Exception:
```

Ruff (0.16.2)

[warning] Do not catch blind exception: `Exception`

(BLE001)